js/checkbox-wikirule.js

```diff
@@ -79,50 +113,80 @@ exports.parse = function() {
                 this.parser.source.substring(startPos + 4, this.parser.pos),
                 {parseAsInline: true});
 
-        // Put the listitem body in a span for easy reference
-        var itembody = {
-            type: "element",
-            tag: "span",
-            attributes: {
-                class: {type: "string", value: "checklistitem-body"}
-            },
-            children: parseResults.tree
-        };
-
+        // Use the listitem body as a label for the checkbox to get better accessibility
         var checkbox = {
             type: "element",
             tag: "input",
             attributes: {
                 type: {type: "string", value: "checkbox"},
-                pos: {type: "string", value: match.index}
+                pos: {type: "string", value: match.index},
+                id: {type: "string", value: match.index}
             }
         };
         if (match[1] === "x" || match[1] === "X") {
             checkbox.attributes.checked = {type: "boolean", value: true};
-            if (this.shouldStrikeChecked()) {
-                itembody.attributes.class.value += ", checkedchecklistitem-body";
-            }
         }
 
-        var removeicon = {
+        var itembody = {
+            type: "element",
+            tag: "label",
+            attributes: {
+                class: {type: "string", value: "checklistitem-body"},
+                for: {type: "string", value: match.index}
+            },
+            children: parseResults.tree
+        };
+
+        // Make a button to delete the item
+        // TODO: use svg bin icon, pattern as follows?
+/*
+  <svg style="display: none">
+    <symbol id="bin-icon" viewBox="0 0 20 20">
+      <path d="[path data here]">
+    </symbol>
+  </svg>
+…
+<button aria-label="delete">
+  <svg>
+    <use xlink:href="#bin-icon"></use>
+  </svg>
+</button>
+*/
+        var removelabel = {
+            type: "element",
+            tag: "span",
+            attributes: {
+                class: {type: "string", value: "checklist-vh"},
+// the next line does nothing, it works here but nowhere else :–(
```

Comment:
I thought this could be a clue for a more elegant solution using (translated) standard texts … but if you don’t know how to make this work either, then I will remove it.